### ISSUE TITLE
Failed to import - @twind/core

### ISSUE THREAD

**Issue by KyleJune** [ije/esm.sh#474]
## Failing module

- **GitHub**: https://github.com/tw-in-js/twind/tree/%40twind/core%401.1.1/packages/core
- **npm**: https://www.npmjs.com/package/@twind/core

```js
import * as twind from "https://esm.sh/@twind/core@1.1.1?target=deno&pin=v99&dev";
```

## Problem

twind is working on the server side but not the client side. I found the cause being an assumption that typeof document == "undefined" will always be true during transformation. The server uses virtual sheets and the browser is suppose to use cssom or dom sheets. You can see that in the following function.

https://github.com/tw-in-js/twind/blob/v1.0.0-next.38/packages/twind/src/sheets.ts#L154
```ts
/**
 * Returns a sheet useable in the current environment.
 *
 * @param useDOMSheet usually something like `process.env.NODE_ENV != 'production'` (default: browser={@link cssom}, server={@link virtual})
 * @param disableResume to not include or use resume data
 * @returns a sheet to use
 */
export function getSheet(
  useDOMSheet?: boolean,
  disableResume?: boolean,
): Sheet<string[] | HTMLStyleElement | CSSStyleSheet> {
  const sheet =
    typeof document == 'undefined' ? virtual(!disableResume) : useDOMSheet ? dom() : cssom()

  if (!disableResume) sheet.resume = resume

  return sheet
}
```

esm.sh appears to assume the first condition will always be true and end up transforming that function into this one.

https://esm.sh/v100/@twind/core@1.1.1/deno/core.development.js
```js
function getSheet(useDOMSheet, disableResume) {
  let sheet = virtual(!disableResume);
  return disableResume || (sheet.resume = resume), sheet;
}
```

This completely changes the behavior of this function and results in cssom and dom never getting called.

## Additional info

- **esm.sh version**: v99 and v100.
- **Deno version**: 1.29.1

**Comment by KyleJune** [comment]
It looks like this isn't a problem if I change the target to es2022.

https://esm.sh/v100/@twind/core@1.1.1/es2022/core.development.js
```js
function getSheet(useDOMSheet, disableResume) {
  let sheet = "undefined" == typeof document ? virtual(!disableResume) : useDOMSheet ? dom() : cssom();
  return disableResume || (sheet.resume = resume), sheet;
}
```

Should I be using es2022 as the target in my deno project for all code used in the client too? One concern I have with that is that I have both browser and server dependencies on react. For example, I use react-helmet-async on both the server and browser and it depends on react. If I used es2022 for react-helmet-async, I believe I would end up with 2 different versions of react on my server, react es2022 and react deno. Here is the react-helmet-async module import I'm referring to.

```ts
export {
  Helmet,
  HelmetProvider,
} from "https://esm.sh/react-helmet-async@1.3.0?target=deno&pin=v99&deps=react@18.2.0,react-dom@18.2.0";
```

Ideally, I should be able to just use the deno target esm.sh modules for both my server and client so that I can do ssr. I have esbuild for bundling my client code with an esm target.

**Comment by KyleJune** [comment]
For a workaround, I'm going to try creating 2 separate import maps. One for when I'm building my client that will use es2022 for the target, and then my server side import map will use deno for the target.

**Comment by ije** [comment]
> It looks like this isn't a problem if I change the target to es2022.

it's weird, since the `@twind/core` doesn't have any deps, the build output should be the same (deno target will use `esnext` as the esbuild option)

**Comment by KyleJune** [comment]
Oh from the twind discord, the author responded to me saying it wasn't esm.sh.

> Regarding the esm issue: That is not esm.sh doing the transformation. That already happened within twinds build pipeline during publishing. Twind creates several bundles for specific use case (node, bundlers, direct browser usage, … and deno). The deno bundle created by twind is the problem here because it doesn’t include the dom sheets.

**Comment by ije** [comment]
> Oh from the twind discord, the author responded to me saying it wasn't esm.sh.
> 
> > Regarding the esm issue: That is not esm.sh doing the transformation. That already happened within twinds build pipeline during publishing. Twind creates several bundles for specific use case (node, bundlers, direct browser usage, … and deno). The deno bundle created by twind is the problem here because it doesn’t include the dom sheets.

interesting...

**Comment by KyleJune** [comment]
I'm going to leave this open for now but I asked the author to change their build pipeline for deno to not do that transformation. Once that is done I'll close this issue. Keeping it open in case anyone runs into a similar issue trying to use twind v1.

**Comment by KyleJune** [comment]
This issue is resolved in twind 1.0.6.
